src/js/components/component.js

```diff
@@ -22,6 +22,7 @@ export default class Component extends Data {
     this.name = name
 
     this.config = Components[`${this.name}s`].config
+    merge(this.config, data.config)
```

Comment:
Good catch. I thought this was handled by the setter here: https://github.com/allan-ayamah/formeo/blob/patch-1/src/js/components/component-data.js#L62. Will double check and merge your PR.